autowarefoundation/autoware.universe
# feat(surround_obstacle_checker): separate surround_obstacle_checker from hierarchical planning flow

## Description
The function of `surround_obstacle_checker` is to keep the stopped state when the vehicle is stopped and there are obstacles around the vehicle. In current implementation, the feature is achieved by inserting ZERO velocity into trajectory in the hierarchical motion planning flow.
On the other hand, the feature is alse achieved by using external velocity limit with simpler implementation. (no need to insert stop point)

In this PR,

- use external velocity limit in `surround_obstacle_checker`
- separate `surround_obstacle_checker` from hierarchical planning flow

https://user-images.githubusercontent.com/44889564/165709891-5db5a0bd-c80a-4ce9-bd46-d98c60309f1e.mp4


## Related links
- https://github.com/autowarefoundation/autoware.universe/issues/832


## Tests performed

1. checkout launcher branch to feature/surround-obstacle-checker (https://github.com/tier4/autoware_launch/pull/304)
2. launch planning_simulator.launch.xml.
3. put GOAL.
4. put obstacke around vehicle.
5. engage and check whether Ego does NOT move.
6. remove obstacle and check whether Ego moves toward the GOAL after few seconds.



## Notes for reviewers

- change event driven to timer driven. (default: 10Hz)



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
